### PR TITLE
add thunderbird module

### PR DESCRIPTION
**Pull request by jkarlson**
### Description

add thunderbird module that creates user.js and profiles.ini
much of this is copied from firefox module

### Checklist

- [ x] Change is backwards compatible.

- [ x] Code formatted with `./format`.

- [ x] Code tested through `nix-shell --pure tests -A run.all`.

- [  ] Test cases updated/added. See [example](https://github.com/nix-community/home-manager/commit/f3fbb50b68df20da47f9b0def5607857fcc0d021#diff-b61a6d542f9036550ba9c401c80f00ef).

- [ x] Commit messages are formatted like

    ```
    {component}: {description}

    {long description}
    ```

    See [CONTRIBUTING](https://github.com/nix-community/home-manager/blob/master/docs/contributing.adoc#sec-commit-style) for more information and [recent commit messages](https://github.com/nix-community/home-manager/commits/master) for examples.

- If this PR adds a new module

  - [x] Added myself as module maintainer. See [example](https://github.com/nix-community/home-manager/blob/068ff76a10e95820f886ac46957edcff4e44621d/modules/programs/lesspipe.nix#L6).

  - [x] Added myself and the module files to `.github/CODEOWNERS`.

### CODE REVIEW

File: modules/programs/thunderbird.nix [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+
+      ${profile.extraConfig}
+    '';
+
+in {
+  meta.maintainers = [ maintainers.jkarlson ];
+
+  options = {
+    programs.thunderbird = {
+      enable = mkEnableOption "Firefox";
```
Comment (polykernel) [inline modules/programs/thunderbird.nix:63]:
```suggestion
      enable = mkEnableOption "Thunderbird, a free, open-source, cross-platform application for managing email, news feeds, chat, and news groups";
```

File: modules/programs/thunderbird.nix [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+    };
+
+  profilesIni = generators.toINI { } profiles;
+
+  mkUserJs = profile:
+    let
+      prefs = lib.optionalAttrs profile.disableTelemetry {
+        "datareporting.healthreport.uploadEnabled" = false;
+        "datareporting.policy.dataSubmissionPolicyAcceptedVersion" = 1;
+        "datareporting.policy.dataSubmissionPolicyNotifiedTime" = "1631721040030";
```
Comment (polykernel) [inline modules/programs/thunderbird.nix:36]: I think this is just some internal data used for logging, and shouldn't be set here. correct me if I am wrong though.

Comment (jkarlson) [reply modules/programs/thunderbird.nix]: It seems that L36 was not required, I assumed you would get the privacy notification without it.

Comment (polykernel) [reply modules/programs/thunderbird.nix]: it is probably better to let the application manage it, I think it is better for options to be minimal so they can be composed.

Comment (polykernel) [reply modules/programs/thunderbird.nix]: A small nitpick, you can ignore it if you disagree, but I think you can have the merge done within the module system using mkIf, like
```nix
settings = mkIf (profile.disableTelemetry) { ... };
```
to take advantage of the merging capability of the module system, but it isn't necessary in this case as there aren't special rules for the set values.

Comment (jkarlson) [reply modules/programs/thunderbird.nix]: can you advise on doing this without circular deps, I don't seem to be able to, as the disableTelemetry resides inside the same set as settings.

Comment (rycee) [reply modules/programs/thunderbird.nix]: I think `optionalAttrs` is best in this case since it will be applied regardless of priority level.

**Comment by rycee** [comment]
Thanks for the contribution! Looks good, I think.

If it is possible to configure smtp and imap through `user.js` then perhaps it would be worth trying to integrate with the email infrastructure?

File: modules/programs/thunderbird.nix (hunk 2 above)
Comment (polykernel) [reply modules/programs/thunderbird.nix]: @jkarlson 

> can you advise on doing this without circular deps, I don't seem to be able to, as the disableTelemetry resides inside the same set as settings.

I apologize for the extremely late response, I think we can push this to the submodule config itself, so something like
```nix
{config, name, ...}:

{
  options = {
    # code
  };
   
  config = {
    settings = mkIf (config.disableTelemetry) {
      # code
    };
  };
}
```

**Comment by d-dervishi** [comment]
I wrote a PR at #3253 to support both profiles and `accounts.email.accounts` configuration options.

**Comment by jkarlson** [comment]
Maybe we can merge 3253

**Pull request closed without merging**